packages/react-router/src/route.ts

```diff
@@ -1083,12 +1083,11 @@ export class Route<
     this._ssr = options?.ssr ?? opts.defaultSsr ?? true
   }
 
-  addChildren<
-    const TNewChildren extends
-      | Record<string, AnyRoute>
-      | ReadonlyArray<AnyRoute>,
-  >(
-    children: TNewChildren,
+  addChildren<const TNewChildren>(
```

Comment:
This constraint was actually causing a recursive performance issue when comparing against `AnyRoute`. I have a partial understanding as to why. But `Constrain` does not have this issue. Starting to give up constraints in many places